pkg/gofr/datasource/file/ftp/file.go

```diff
@@ -301,6 +348,13 @@ func (f *file) Write(p []byte) (n int, err error) {
 
 	f.offset += int64(len(p))
 
+	t := time.Time{}
+
+	mt := f.ModTime()
+	if mt != t {
+		f.modTime = mt
+	}
```

Comment:
Is this what you were looking for?

```suggestion
	mt := f.ModTime()
	if !mt.IsZero() {
		f.modTime = mt
	}
```